### PR TITLE
Convert module to typescript

### PR DESCRIPTION
**Pull request by kellhogs**
I've converted this module to typescript in preparation of the Ontime v2 api update

### CODE REVIEW

**Comment by cpvalente** [comment]
This looks good from app side, good work
@josephdadams  or @Julusian  would you guys be available for some sanity check?

Also, is it possible to add collaborator permissions to @kellhogs , he is maintaining the module on its own at this moment

File: package.json [hunk 1]
```diff
@@ -1,25 +1,50 @@
...
-		"lint": "yarn lint:raw ."
+		"format": "prettier -w .",
+		"postinstall": "yarn build"
+	},
+	"engines": {
+		"node": ">=18.12.0"
 	},
 	"dependencies": {
-		"@companion-module/base": "~1.0.0",
+		"@companion-module/base": "^1.2.1",
```
Comment (dnmeid) [inline package.json:26]: Not a good idea to get base with ^, you should use the lowest version you really need with ~
With ^1.2.1 you currently get 1.4.0 and that is not compatible with the beta branch of Companion 3.0

**Review by dnmeid — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (kellhogs) [reply package.json:26]: good catch that i actually already reverted back to ~1.2.1 for the next pr i have in line. Anyway i will revert it here to.

**Comment by dnmeid** [comment]
I guess you just did the change in source and then did not run yarn again. So currently yarn.lock still is pointing to base v1.4.
The current beta has now been updated to base 1.4 too, so this won't be a problem any more for users running the very latest beta.
Just tell me if you want it to be merged like this or if you want to sync yarn.lock with package.json before.

**Comment by kellhogs** [comment]
I've synced the yarn.lock and it qould be great if you could merge here. I think we dont need to create a new release here tho cause i've already finished the update for the V2 API update. Once you have merged i'll try to rule out any merge conflicts that might turn up and the post the next PR. 

Looking out for what you have to say to it

**Pull request merged**